### PR TITLE
Refactor base.pp

### PR DESCRIPTION
**Pull request by alexjfisher**
Make the way we define conditional dependencies consistent and remove
the, (IMO dubious), use of merging multiple parameters into the splat
attribute.

### CODE REVIEW

File: manifests/base.pp [hunk 1]
```diff
@@ -1,13 +1,5 @@
-# Class: rsyslog::base
-#
-# Description
-# -----------
-#
-# This class manages the base installation for rsyslog
+# @summary This class manages the base installation for rsyslog
 class rsyslog::base {
-  # Include the base class in case this class is being called
```
Comment (alexjfisher) [inline manifests/base.pp:8]: This comment doesn't seem to like it belongs here anymore

File: manifests/base.pp [hunk 2]
```diff
@@ -74,15 +56,14 @@
       # all configuration is under the rsyslog.d directory
       |EOT
 
-    $_require = $rsyslog::manage_package ? {
-      false   => undef,
-      default => Package[$rsyslog::package_name],
-    }
     file { $rsyslog::config_file:
-      ensure  => 'file',
+      ensure  => file,
```
Comment (alexjfisher) [inline manifests/base.pp:60]: Apparently the style guide says to do this.  I don't really care, but at least it's consistent with how `ensure => directory` doesn't single quote `directory` in the rest of this manifest.  (That and vim puppet highlights the bare words `file` and `directory`!)

File: manifests/base.pp [hunk 3]
```diff
@@ -74,15 +56,14 @@
       # all configuration is under the rsyslog.d directory
       |EOT
 
-    $_require = $rsyslog::manage_package ? {
```
Comment (alexjfisher) [inline manifests/base.pp:77]: This was much better than merging into `*`, but I've updated it to keep everything consistent.

File: manifests/base.pp [hunk 4]
```diff
@@ -44,27 +36,17 @@
...
-      true => {
-        'require' => Package[$rsyslog::package_name],
-      },
-      false => {}
+    file { $rsyslog::confdir:
+      ensure  => directory,
+      owner   => 'root',
+      group   => 'root',
+      mode    => $rsyslog::confdir_permissions,
+      purge   => $rsyslog::purge_config_files,
```
Comment (alexjfisher) [inline manifests/base.pp:44]: Note `purge` and `recurse` both default to `false` in file resources, so this shouldn't change behaviour.

**Comment by alexjfisher** [comment]
Replaces https://github.com/voxpupuli/puppet-rsyslog/pull/178

**Review by kenyon — APPROVED** [review]

File: manifests/base.pp [hunk 5]
```diff
@@ -1,13 +1,5 @@
-# Class: rsyslog::base
-#
-# Description
-# -----------
-#
-# This class manages the base installation for rsyslog
+# @summary This class manages the base installation for rsyslog
```
Comment (ekohl) [inline manifests/base.pp:1]: Is this also a private class or is it public?

**Review by ekohl — APPROVED** [review]
Logically this looks the same, but it's much easier to follow. :+1:

File: manifests/base.pp [hunk 6]
```diff
@@ -44,27 +36,17 @@
...
     }
 
-    file { $rsyslog::confdir:
-      ensure => directory,
-      owner  => 'root',
-      group  => 'root',
-      mode   => $rsyslog::confdir_permissions,
-      *      => $purge_params + $require_package,
+    if $rsyslog::manage_package {
+      Package[$rsyslog::package_name] -> File[$rsyslog::confdir]
```
Comment (ekohl) [inline manifests/base.pp:49]: I see there was some discussion about this vs meta parameters. You can consider the following:
```puppet
$package_require = $rsyslog::manage_package ? {
  true => Package[$rsyslog::package_name],
  default => undef,
}

file { $file:
  require => $package_require,
}
```

I'm not sure if it's better. If there's one case, I'd say it's not. In this case there are 2 files using it which makes a better case.

Both are certainly better than what exists now with `* => $require_package`. That's just <del>magic</del> hard to follow and thus easy to miss.

File: manifests/base.pp (hunk 5 above)
Comment (dhollinger) [reply manifests/base.pp:1]: Should probably be private since it's not directly called by another class

**Review by dhollinger — APPROVED** [review]
